--- gcc/d/Make-lang.in	2017-09-30 00:27:44.232360842 +0200
+++ gcc/d/Make-lang.in	2017-09-30 00:27:56.543360988 +0200
@@ -92,10 +92,10 @@
 
 # Generated programs.
 d/idgen: d/idgen.dmdgen.o
-	+$(LINKER_FOR_BUILD) $(BUILD_LINKER_FLAGS) $(BUILD_LDFLAGS)  -o $@ $^
+	+$(LINKER_FOR_BUILD) $(BUILD_LINKER_FLAGS) $(BUILD_LDFLAGS)  -o $@ $^ -lgcov
 
 d/impcvgen: d/impcnvgen.dmdgen.o
-	+$(LINKER_FOR_BUILD) $(BUILD_LINKER_FLAGS) $(BUILD_LDFLAGS) -o $@ $^
+	+$(LINKER_FOR_BUILD) $(BUILD_LINKER_FLAGS) $(BUILD_LDFLAGS) -o $@ $^ -lgcov
 
 # Generated sources.
 d/id.c: d/idgen
