--- GNUmakefile
+++ GNUmakefile
@@ -26,14 +25,12 @@
 
 all: libs shared-libs gda gstreamer vte shared-gda shared-gstreamer shared-vte test
 
-ifeq ("$(DC)","gdc")
-    DCFLAGS=-O2
+ifeq ("$(DLANG_VENDOR)","gdc")
     LINKERFLAG=-Xlinker 
     DDOCFLAGS=-fsyntax-only -c -fdoc -fdoc-file=$@
     DDOCINC=-fdoc-inc=
     output=-o $@
 else
-    DCFLAGS=-O
     LINKERFLAG=-L
     DDOCFLAGS=-o- -Df$@
     output=-of$@
@@ -246,8 +243,8 @@
 	echo Name: GtkD > $@
 	echo Description: A D binding and OO wrapper for GTK+. >> $@
 	echo Version: $(GTKD_VERSION) >> $@
-	echo Libs: $(LINKERFLAG)-L$(prefix)/lib/ $(LINKERFLAG)-lgtkd-$(MAJOR) $(LINKERFLAG)-ldl >> $@
-	echo Cflags: -I$(prefix)/include/d/gtkd-$(MAJOR)/ >> $@
+	echo Libs: $(LINKERFLAG)-lgtkd-$(MAJOR) $(LINKERFLAG)-ldl >> $@
+	echo Cflags: -I$(DLANG_IMPORT_DIR)/gtkd-$(MAJOR)/ >> $@
 
 gtkdgl-$(MAJOR).pc:
 	echo Name: GtkDGL > $@
@@ -297,26 +294,26 @@
 endif
 
 install-gtkd: $(LIBNAME_GTKD) install-headers-gtkd
-	install -d $(DESTDIR)$(prefix)/$(libdir)
-	install -m 644 $(LIBNAME_GTKD)   $(DESTDIR)$(prefix)/$(libdir)
+	install -d $(DESTDIR)$(DLANG_LIB_DIR)
+	install -m 644 $(LIBNAME_GTKD)   $(DESTDIR)$(DLANG_LIB_DIR)
 
 install-gtkdgl: $(LIBNAME_GTKDGL) install-gtkd install-headers-gtkdgl
-	install -m 644 $(LIBNAME_GTKDGL) $(DESTDIR)$(prefix)/$(libdir)
+	install -m 644 $(LIBNAME_GTKDGL) $(DESTDIR)$(DLANG_LIB_DIR)
 
 install-gtkdsv: $(LIBNAME_GTKDSV) install-gtkd install-headers-gtkdsv
-	install -m 644 $(LIBNAME_GTKDSV) $(DESTDIR)$(prefix)/$(libdir)
+	install -m 644 $(LIBNAME_GTKDSV) $(DESTDIR)$(DLANG_LIB_DIR)
 
 install-gda: $(LIBNAME_GTKDGDA) install-gtkd install-headers-gda
-	install -m 644 $(LIBNAME_GTKDGDA) $(DESTDIR)$(prefix)/$(libdir)
+	install -m 644 $(LIBNAME_GTKDGDA) $(DESTDIR)$(DLANG_LIB_DIR)
 
 install-gstreamer: $(LIBNAME_GSTREAMERD) install-gtkd install-headers-gstreamer
-	install -m 644 $(LIBNAME_GSTREAMERD) $(DESTDIR)$(prefix)/$(libdir)
+	install -m 644 $(LIBNAME_GSTREAMERD) $(DESTDIR)$(DLANG_LIB_DIR)
 
 install-vte: $(LIBNAME_VTED) install-gtkd install-headers-vte
-	install -m 644 $(LIBNAME_VTED) $(DESTDIR)$(prefix)/$(libdir)
+	install -m 644 $(LIBNAME_VTED) $(DESTDIR)$(DLANG_LIB_DIR)
 
 install-shared-gtkd: $(SONAME_GTKD)
-	install -d $(DESTDIR)$(prefix)/$(libdir)
+	install -d $(DESTDIR)$(DLANG_LIB_DIR)
 	$(install-so)
 
 install-shared-gtkdgl: $(SONAME_GTKDGL) install-shared-gtkd
@@ -335,29 +332,29 @@
 	$(install-so)
 
 install-headers-gtkd: gtkd-$(MAJOR).pc
-	install -d $(DESTDIR)$(prefix)/include/d/gtkd-$(MAJOR)
+	install -d $(DESTDIR)$(DLANG_IMPORT_DIR)/gtkd-$(MAJOR)
 	install -d $(DESTDIR)$(datadir)/pkgconfig
-	(cd src;   echo $(SOURCES_GTKD)   | sed -e s,src/,,g   | xargs tar cf -) | (cd $(DESTDIR)$(prefix)/include/d/gtkd-$(MAJOR); tar xv)
+	(cd src;   echo $(SOURCES_GTKD)   | sed -e s,src/,,g   | xargs tar cf -) | (cd $(DESTDIR)$(DLANG_IMPORT_DIR)/gtkd-$(MAJOR); tar xv)
 	install -m 644 gtkd-$(MAJOR).pc $(DESTDIR)$(datadir)/pkgconfig
 
 install-headers-gtkdgl: gtkdgl-$(MAJOR).pc install-headers-gtkd
-	(cd srcgl; echo $(SOURCES_GTKDGL) | sed -e s,srcgl/,,g | xargs tar cf -) | (cd $(DESTDIR)$(prefix)/include/d/gtkd-$(MAJOR); tar xv)
+	(cd srcgl; echo $(SOURCES_GTKDGL) | sed -e s,srcgl/,,g | xargs tar cf -) | (cd $(DESTDIR)$(DLANG_IMPORT_DIR)/gtkd-$(MAJOR); tar xv)
 	install -m 644 gtkdgl-$(MAJOR).pc $(DESTDIR)$(datadir)/pkgconfig
 
 install-headers-gtkdsv: gtkdsv-$(MAJOR).pc install-headers-gtkd
-	(cd srcsv; echo $(SOURCES_GTKDSV) | sed -e s,srcsv/,,g | xargs tar cf -) | (cd $(DESTDIR)$(prefix)/include/d/gtkd-$(MAJOR); tar xv)
+	(cd srcsv; echo $(SOURCES_GTKDSV) | sed -e s,srcsv/,,g | xargs tar cf -) | (cd $(DESTDIR)$(DLANG_IMPORT_DIR)/gtkd-$(MAJOR); tar xv)
 	install -m 644 gtkdsv-$(MAJOR).pc $(DESTDIR)$(datadir)/pkgconfig
 
 install-headers-gda: gtkdgda-$(MAJOR).pc install-headers-gtkd
-	(cd srcgda; echo $(SOURCES_GTKDGDA) | sed -e s,srcgda/,,g | xargs tar cf -) | (cd $(DESTDIR)$(prefix)/include/d/gtkd-$(MAJOR); tar xv)
+	(cd srcgda; echo $(SOURCES_GTKDGDA) | sed -e s,srcgda/,,g | xargs tar cf -) | (cd $(DESTDIR)$(DLANG_IMPORT_DIR)/gtkd-$(MAJOR); tar xv)
 	install -m 644 gtkdgda-$(MAJOR).pc $(DESTDIR)$(datadir)/pkgconfig
 
 install-headers-gstreamer: gstreamerd-$(MAJOR).pc install-headers-gtkd
-	(cd srcgstreamer; echo $(SOURCES_GSTREAMERD) | sed -e s,srcgstreamer/,,g | xargs tar cf -) | (cd $(DESTDIR)$(prefix)/include/d/gtkd-$(MAJOR); tar xv)
+	(cd srcgstreamer; echo $(SOURCES_GSTREAMERD) | sed -e s,srcgstreamer/,,g | xargs tar cf -) | (cd $(DESTDIR)$(DLANG_IMPORT_DIR)/gtkd-$(MAJOR); tar xv)
 	install -m 644 gstreamerd-$(MAJOR).pc $(DESTDIR)$(datadir)/pkgconfig
 
 install-headers-vte: vted-$(MAJOR).pc install-headers-gtkd
-	(cd srcvte; echo $(SOURCES_VTED) | sed -e s,srcvte/,,g | xargs tar cf -) | (cd $(DESTDIR)$(prefix)/include/d/gtkd-$(MAJOR); tar xv)
+	(cd srcvte; echo $(SOURCES_VTED) | sed -e s,srcvte/,,g | xargs tar cf -) | (cd $(DESTDIR)$(DLANG_IMPORT_DIR)/gtkd-$(MAJOR); tar xv)
 	install -m 644 vted-$(MAJOR).pc $(DESTDIR)$(datadir)/pkgconfig
 
 uninstall: uninstall-gtkdgl uninstall-gtkdsv uninstall-gda uninstall-gstreamer
@@ -426,13 +423,13 @@
 
 define make-shared-lib
 	#Remove this line when phobos #1280 is merged.
-	$(if $(findstring "dmd","$(DC)"),$(eval LDFLAGS+=-defaultlib=:libphobos2.so))
+	$(if $(findstring "dmd","$(DLANG_VENDOR)"),$(eval LDFLAGS+=-defaultlib=:libphobos2.so))
  
 	$(DC) -shared $(output) $(LDFLAGS) $(LINKERFLAG)-soname=$@.$(SO_VERSION) $^
 endef
 
 define install-so
-    install -m 755 $< $(DESTDIR)$(prefix)/$(libdir)/$<.$(SO_VERSION).$(MINOR).$(BUGFIX)
-    cd $(DESTDIR)$(prefix)/$(libdir)/; ln -s $<.$(SO_VERSION).$(MINOR).$(BUGFIX) $<.$(SO_VERSION) 
-    cd $(DESTDIR)$(prefix)/$(libdir)/; ln -s $<.$(SO_VERSION) $<
+    install -m 755 $< $(DESTDIR)$(DLANG_LIB_DIR)/$<.$(SO_VERSION).$(MINOR).$(BUGFIX)
+    cd $(DESTDIR)$(DLANG_LIB_DIR)/; ln -s $<.$(SO_VERSION).$(MINOR).$(BUGFIX) $<.$(SO_VERSION) 
+    cd $(DESTDIR)$(DLANG_LIB_DIR)/; ln -s $<.$(SO_VERSION) $<
 endef
