--- a/runtime/druntime/src/ldc/intrinsics.di
+++ b/runtime/druntime/src/ldc/intrinsics.di
@@ -61,6 +61,15 @@
     version = INTRINSICS_FROM_600;
     version = INTRINSICS_FROM_700;
 }
+else version (LDC_LLVM_701)
+{
+    version = INTRINSICS_FROM_308;
+    version = INTRINSICS_FROM_309;
+    version = INTRINSICS_FROM_400;
+    version = INTRINSICS_FROM_500;
+    version = INTRINSICS_FROM_600;
+    version = INTRINSICS_FROM_700;
+}
 else
 {
     static assert(false, "LDC LLVM version not supported");
