--- a/utils/CMakeLists.txt	2020-02-10 16:40:25.516183683 +0100
+++ b/utils/CMakeLists.txt	2020-02-10 16:37:23.941181536 +0100
@@ -31,7 +31,7 @@
     target_link_libraries(gen_gccbuiltins dl)
 endif()

-if (TARGET FileCheck)
+if ((TARGET FileCheck) OR (EXISTS ${LLVM_ROOT_DIR}/bin/FileCheck))
   # already provided by LLVM
 else()
 # Build FileCheck for testing (build source version depending on LLVM version)
@@ -50,7 +50,7 @@
 endif()
 endif()

-if (TARGET not)
+if ((TARGET not) OR (EXISTS ${LLVM_ROOT_DIR}/bin/not))
   # already provided by LLVM
 else()
 # Build `not` for testing
