--- a/runtime/CMakeLists.txt
+++ b/runtime/CMakeLists.txt
@@ -252,7 +252,7 @@

 # Only have either shared or static libs?
 # Then explicitly default to linking against them via default LDC switch.
-if(${BUILD_SHARED_LIBS} STREQUAL "ON")
+if(NOT ${BUILD_SHARED_LIBS} STREQUAL "OFF")
     set(ADDITIONAL_DEFAULT_LDC_SWITCHES "${ADDITIONAL_DEFAULT_LDC_SWITCHES}\n        \"-link-defaultlib-shared\",")
 elseif(${BUILD_SHARED_LIBS} STREQUAL "OFF")
     set(ADDITIONAL_DEFAULT_LDC_SWITCHES "${ADDITIONAL_DEFAULT_LDC_SWITCHES}\n        \"-link-defaultlib-shared=false\",")
