--- a/dmd2/root/port.c
+++ b/dmd2/root/port.c
@@ -433,7 +433,7 @@
 
 #if __linux__ || __APPLE__ || __FreeBSD__ || __OpenBSD__ || __HAIKU__
 
-#include <math.h>
+#include <cmath>
 #if __linux__
 #include <bits/nan.h>
 #include <bits/mathdef.h>
@@ -581,7 +581,7 @@
     return isnan(r);
 #else
     #undef isnan
-    return ::isnan(r);
+    return std::isnan(r);
 #endif
 }
 
@@ -597,7 +597,7 @@
     return isnan(r);
 #else
     #undef isnan
-    return ::isnan(r);
+    return std::isnan(r);
 #endif
 }
 
@@ -625,16 +625,16 @@
     return isinf(r);
 #else
     #undef isinf
-    return ::isinf(r);
+    return std::isinf(r);
 #endif
 }
 
 longdouble Port::fmodl(longdouble x, longdouble y)
 {
 #if __FreeBSD__ && __FreeBSD_version < 800000 || __OpenBSD__
     return ::fmod(x, y);        // hack for now, fix later
 #else
-    return ::fmodl(x, y);
+    return std::fmod(x, y);
 #endif
 }
 
