diff --git a/dmd2/func.c b/dmd2/func.c
index e7ec19e..9b16322 100644
--- a/dmd2/func.c
+++ b/dmd2/func.c
@@ -4174,9 +4174,14 @@ void FuncLiteralDeclaration::toCBuffer(OutBuffer *buf, HdrGenState *hgs)
 
     CompoundStatement *cs = fbody->isCompoundStatement();
     Statement *s1;
+#if IN_LLVM // backport from DMD 2.066
+    if (semanticRun >= PASSsemantic3done && cs)
+    {
+#else
     if (semanticRun >= PASSsemantic3done)
     {
         assert(cs);
+#endif
         s1 = (*cs->statements)[cs->statements->dim - 1];
     }
     else
