diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8ba5ffd..31af935 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -34,7 +34,7 @@ include(GetLinuxDistribution)
 #
 
 # Version information
-set(LDC_VERSION "0.15.0") # May be overridden by git hash tag
+set(LDC_VERSION "0.15.1") # May be overridden by git hash tag
 set(DMDFE_MAJOR_VERSION   2)
 set(DMDFE_MINOR_VERSION   0)
 set(DMDFE_PATCH_VERSION   66)
