diff --git a/source/dub/generators/build.d b/source/dub/generators/build.d
--- a/source/dub/generators/build.d
+++ b/source/dub/generators/build.d
@@ -119,10 +119,10 @@ class BuildGenerator : ProjectGenerator {
 		auto build_id = computeBuildID(config, buildsettings, settings);
 
 		// make all paths relative to shrink the command line
-		string makeRelative(string path) { return shrinkPath(NativePath(path), cwd); }
+		/+string makeRelative(string path) { return shrinkPath(NativePath(path), cwd); }
 		foreach (ref f; buildsettings.sourceFiles) f = makeRelative(f);
 		foreach (ref p; buildsettings.importPaths) p = makeRelative(p);
-		foreach (ref p; buildsettings.stringImportPaths) p = makeRelative(p);
+		foreach (ref p; buildsettings.stringImportPaths) p = makeRelative(p);+/
 
 		// perform the actual build
 		bool cached = false;
