--- dmd2/dmd/src/posix.mak
+++ dmd2/dmd/src/posix.mak
@@ -76,6 +76,9 @@ endif

 CFLAGS = $(GFLAGS) -I$(ROOT) -DMARS=1 -DTARGET_$(OS)=1 -DDM_TARGET_CPU_$(TARGET_CPU)=1
 MFLAGS = $(GFLAGS) -I$C -I$(TK) -I$(ROOT) -DMARS=1 -DTARGET_$(OS)=1 -DDM_TARGET_CPU_$(TARGET_CPU)=1 -DDMDV2=1
+ifeq ($(findstring g++,$(HOST_CC)), g++)
+MFLAGS := $(MFLAGS) -std=gnu++98
+endif

 CH= $C/cc.h $C/global.h $C/oper.h $C/code.h $C/type.h \
 	$C/dt.h $C/cgcv.h $C/el.h $C/obj.h $(TARGET_CH)
