--- dmd2/src/dmd/posix.mak	2014-02-24 05:23:02.000000000 +0100
+++ dmd2/src/dmd/posix.mak	2014-03-15 00:48:37.568627267 +0100
@@ -649,7 +649,7 @@ install: all
 	mkdir -p $(INSTALL_DIR)/bin
 	cp dmd $(INSTALL_DIR)/bin/dmd
 	$(eval bin_dir=$(if $(filter $(OS),osx), bin, bin$(MODEL)))
-	cp ../ini/$(OS)/$(bin_dir)/dmd.conf $(INSTALL_DIR)/bin/dmd.conf
+	cp dmd.conf.default $(INSTALL_DIR)/bin/dmd.conf
 	cp backendlicense.txt $(INSTALL_DIR)/dmd-backendlicense.txt
 	cp artistic.txt $(INSTALL_DIR)/dmd-artistic.txt
 
--- dmd2/src/druntime/posix.mak	2014-02-24 05:23:06.000000000 +0100
+++ dmd2/src/druntime/posix.mak	2014-03-14 22:32:23.124022918 +0100
@@ -251,8 +251,8 @@ install: target
 	cp -r doc/* $(INSTALL_DIR)/html/
 	mkdir -p $(INSTALL_DIR)/import
 	cp -r import/* $(INSTALL_DIR)/import/
-	mkdir -p $(INSTALL_DIR)/lib
-	cp -r lib/* $(INSTALL_DIR)/lib/
+	mkdir -p $(INSTALL_DIR)/$(LIB_DIR)
+	cp -r lib/libdruntime-linux$(MODEL)* $(INSTALL_DIR)/$(LIB_DIR)/
 	cp LICENSE $(INSTALL_DIR)/druntime-LICENSE.txt
 
 clean: $(addsuffix /.clean,$(ADDITIONAL_TESTS))
--- dmd2/src/phobos/posix.mak	2014-02-24 05:23:10.000000000 +0100
+++ dmd2/src/phobos/posix.mak	2014-03-15 00:03:45.274411999 +0100
@@ -108,7 +108,7 @@ ifeq ($(OS),win32wine)
 	DMD = wine dmd.exe
 	RUN = wine
 else
-	DMD = ../dmd/src/dmd
+	DMD = ../dmd/dmd
 	ifeq ($(OS),win32)
 		CC = dmc
 	else
@@ -354,10 +354,10 @@ zip :
 	zip $(ZIPFILE) $(MAKEFILE) $(ALL_D_FILES) $(ALL_C_FILES) win32.mak win64.mak
 
 install2 : release
-	mkdir -p $(INSTALL_DIR)/lib
-	cp $(LIB) $(INSTALL_DIR)/lib/
+	mkdir -p $(INSTALL_DIR)/$(LIB_DIR)
+	cp $(LIB) $(INSTALL_DIR)/$(LIB_DIR)/
 ifneq (,$(findstring $(OS),linux))
-	cp -P $(LIBSO) $(ROOT)/$(SONAME) $(ROOT)/libphobos2.so $(INSTALL_DIR)/lib/
+	cp -P $(LIBSO) $(ROOT)/$(SONAME) $(ROOT)/libphobos2.so $(INSTALL_DIR)/$(LIB_DIR)/
 endif
 	mkdir -p $(INSTALL_DIR)/import/etc
 	mkdir -p $(INSTALL_DIR)/import/std
