--- dmd2/dmd/src/tk/mem.c
+++ dmd2/dmd/src/tk/mem.c
@@ -755,8 +755,8 @@ void *mem_fmalloc(size_t numbytes)
 {   void *p;
 
     //printf("fmalloc(%d)\n",numbytes);
-#if defined(__llvm__) && (defined(__GNUC__) || defined(__clang__))
-    // LLVM-GCC and Clang assume some types, notably elem (see DMD issue 6215),
+#if defined(__GNUC__) || defined(__clang__)
+    // GCC and Clang assume some types, notably elem (see DMD issue 6215),
     // to be 16-byte aligned. Because we do not have any type information
     // available here, we have to 16 byte-align everything.
     numbytes = (numbytes + 0xF) & ~0xF;
