--- dmd2/dmd/src/dmd/link.d
+++ dmd2/dmd/src/dmd/link.d
@@ -446,6 +446,8 @@         else version (Posix)
         {
             if (global.params.dll)
                 argv.push("-shared");
+                if (!target.is64bit)
+                    argv.push("-fuse-ld=bfd");
         }
         // None of that a.out stuff. Use explicit exe file name, or
         // generate one from name of first source file.
